### PR TITLE
feat(authenticate): add approle auth method

### PR DESCRIPTION
**Pull request by Ameausoone**

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -2875,7 +2875,7 @@ module.exports = function (obj) {
...
...ss-response":false,"electron":false},"bugs":{"url":"https://github.com/sindresorhus/got/issues"},"dependencies":{"@sindresorhus/is":"^0.14.0","@szmarczak/http-timer":"^1.1.2","cacheable-request":"^6.0.0","decompress-response":"^3.3.0","duplexer3":"^0.1.4","get-stream":"^4.1.0","lowercase-keys":"^1.0.1","mimic-response":"^1.0.1","p-cancelable":"^1.0.0","to-readable-stream":"^1.0.0","url-parse-lax":"^3.0.0"},"description":"Simplified HTTP requests","devDependencies":{"ava":"^1.1.0","coveralls":"^3.0.0","delay":"^4.1.0","form-data":"^2.3.3","get-port":"^4.0.0","np":"^3.1.0","nyc":"^13.1.0","p-event":"^2.1.0","pem":"^1.13.2","proxyquire":"^2.0.1","sinon":"^7.2.2","slow-stream":"0.0.4","tempfile":"^2.0.0","tempy":"^0.2.1","tough-cookie":"^3.0.0","xo":"^0.24.0"},"engines":{"node":">=8.6"},"files":["source"],"homepage":"https://github.com/sindresorhus/got#readme","keywords":["http","https","get","got","url","uri","request","util","utility","simple","curl","wget","fetch","net","network","electron"],"license":"MIT","main":"source","name":"got","repository":{"type":"git","url":"git+https://github.com/sindresorhus/got.git"},"scripts":{"release":"np","test":"xo && nyc ava"},"version":"9.6.0"};
```
Comment (devon-kim) [inline dist/index.js]: `/Users/20012243/Dev/vault-action` is this intended?

Comment (Ameausoone) [reply dist/index.js]: Hello @devon-kim thx for reviewing, 

tl;dr : absolutely not.

I am very far from an expert in nodejs stuff, and I didn't know what dist/index.js was, and I just kept @sfauvart commit as is. 

A colleague just explain me it is a compiled version of action.js with `npm run build`.

If I run `build`, I have a similar result with a `...[["got@9.6.0","/Users/antoinemeausoone/Projects/...`, so I don't know how to have a clean compiled index.js ?

**Comment by RichiCoder1** [comment]
Howdy! I'll be taking a look at this here tomorrow FYI. Broadly LGTM, may have some questions and comments.

File: dist/index.js (hunk 1 above)
Comment (RichiCoder1) [reply dist/index.js]: Dist getting regenerated is normal, user paths getting included is not. I'm using [Zeit's ncc tool](https://github.com/zeit/ncc) if you want to check the issues and see if someone else has seen this.

Comment (gVirtu) [reply dist/index.js]: Did you guys figure this out? I was working on a feature and noticed I was getting the same behavior (local path getting injected for some reason) when building. So far couldn't find any additional info in ncc's issues, but I'll keep looking. Thanks!

Comment (Ameausoone) [reply dist/index.js]: OK I updated `got` to `^10.2.2` ,and compiled index.js. Hope it's ok. Thx @gVirtu

Comment (gVirtu) [reply dist/index.js]: There's a tiny caveat though, some tests might need adapting to the new got parameter semantics, instead of:
```
body: <some body>
json: true
```

It now expects something of the form:
```
json: <some body>
```

**Comment by Ameausoone** [comment]
Great work, that's perfect for me !

**Comment by sfauvart** [comment]
👏 Thank you for the fast work and your support !

**Comment by RichiCoder1** [comment]
Apologies for the  delay. Thank you for your contribution!
